src/session.ts

```diff
@@ -147,8 +157,27 @@ export class DebugSession implements IDebugger.ISession {
     return reply.promise;
   }
 
+  /**
+   * Send an execute request message to the kernel.
+   * @param msg execute request message to send to the kernel.
+   */
+  private async _sendExecuteMessage(
+    msg: KernelMessage.IExecuteRequestMsg['content']
+  ): Promise<void> {
+    const kernel = this.client.kernel;
+    if (this._executeFuture) {
+      this._executeFuture.dispose();
+    }
+    this._executeFuture = kernel.requestExecute(msg);
+    this._executeFuture.onReply = (msg: KernelMessage.IExecuteReplyMsg) => {};
+  }
```

Comment:
I am confused about the promise this method is returning, in light of my last comment.

What is this promise guaranteeing when it resolves?

Reply:
This should probably be handled in a better way indeed or simplified.

The idea is that an `execute_request` is sent to the kernel to start the debugging session. But the reply message will be received only when all threads have continued.

Also I would say the reply message (`execute_reply`) can be ignored for now.

Reply:
Actually the `execute_request` should be triggered by a user (ctrl-enter a cell), so the `DebugSession` doesn't need this `execute` method.

In the tests we can trigger an `execute_request` by using the kernel directly:

`client.kernel.requestExecute({ code })`